dev/src/reference.ts

```diff
@@ -1362,10 +1473,50 @@ export class Query<T = firestore.DocumentData> implements firestore.Query<T> {
     fieldPath: string | firestore.FieldPath,
     opStr: firestore.WhereFilterOp,
     value: unknown
+  ): Query<T>;
+
+  /**
+   * Creates and returns a new [Query]{@link Query} with the additional filter
+   * that documents should satisfy the relation constraint provided. Documents
+   * must contain the field specified in the filter.
+   *
+   * Returns a new Query that constrains the value of a Document property.
+   *
+   * This function returns a new (immutable) instance of the Query (rather than
+   * modify the existing instance) to impose the filter.
+   *
+   * @param {Filter} filter A unary or composite filter to apply to the Query.
+   * @returns {Query} The created Query.
+   *
+   * @example
+   * ```
+   * let collectionRef = firestore.collection('col');
+   *
+   * collectionRef.where(Filter.and(Filter.where('foo', '==', 'bar'), Filter.where('foo', '!=', 'baz'))).get()
+   *   .then(querySnapshot => {
+   *     querySnapshot.forEach(documentSnapshot => {
+   *       console.log(`Found document at ${documentSnapshot.ref.path}`);
+   *     });
+   * });
+   * ```
+   *
+   * @private TODO remove private and internal when OR query support is public
+   * @internal
+   */
+  where(filter: Filter): Query<T>;
+
+  where(
```

Comment:
This is the implementation of the two overloads `Query.where(filter)` and `Query.where(field, op, value)`. The implementation needs to match the method name, but this will not be a method documented in API docs or in the `.d.ts` file.